src/types.rs

```diff
@@ -0,0 +1,57 @@
+#[derive(PartialEq, Clone, Copy, Show)]
+pub struct Coordinate {
+    pub x: f64,
+    pub y: f64,
+}
+
+#[derive(PartialEq, Clone, Copy)]
+pub struct Point(pub Coordinate);
+
+#[derive(PartialEq, Clone)]
+pub struct MultiPoint(pub Vec<Point>);
+
+#[derive(PartialEq, Clone)]
+pub struct LineString(pub Vec<Point>);
+
+#[derive(PartialEq, Clone)]
+pub struct MultiLineString(pub Vec<LineString>);
+
+#[derive(PartialEq, Clone)]
+pub struct Polygon(pub LineString, pub Vec<LineString>);
```

Comment:
Because the first LineString (LineRing actually)  in the vector is the exterior ring, the rest are holes. They have different meaning. Also this structure implies that Polygon should at least have one exterior ring, zero or multiple interior rings.